desktop-widgets/locationinformation.cpp

```diff
@@ -313,15 +313,18 @@ void LocationInformationWidget::on_diveSiteDistance_textChanged(const QString &s
 
 void LocationInformationWidget::reverseGeocode()
 {
-	dive_site *ds = diveSite; /* Use local copy in case the object-copy is NULL'd by another thread while we perform the geo lookup */
+	bool buttonWasEnabled = ui.geoCodeButton->isEnabled();
+	ui.geoCodeButton->setEnabled(false);
```

Comment:
This will not work properly, as the `geoCodeButton` is set at different places.
What could happen:
- User starts long running geo lookup. Button is disabled.
- User switches to dive without dive site. Button is disabled.
- Geo lookup finishes. Button is enabled, even if it shouldn't be.

This needs reference counting - I can do it if you want.

Reply:
Ok, sure.  It's kindof out-of-spec for this bugfix/pull-request anyway, and UI code isn't my forte.

Sorry, didn't see this comment until just now (only saw the comments in the main thread) so ignore my latest commit.  I'll revert this pull request to just focus on the bugfix itself.